From: Trond Myklebust <trond.myklebust@primarydata.com>
Date: Mon, 28 Dec 2015 13:30:42 -0500
Subject: [PATCH] NFS: Background flush should not be low priority
Git-commit: b0ac1bd2bbfd5e500432714e55a791c4d394047f
Patch-mainline: v4.5
References: bsc#955308

Background flush is needed in order to satisfy the global page limits.
Don't subvert by reducing the priority.
This should also address a write starvation issue that was reported by
Neil Brown.

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/nfs/write.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/nfs/write.c
+++ b/fs/nfs/write.c
@@ -247,8 +247,6 @@ static int wb_priority(struct writeback_
 		return FLUSH_HIGHPRI | FLUSH_STABLE;
 	if (wbc->sync_mode == WB_SYNC_ALL)
 		ret = FLUSH_COND_STABLE;
-	if (wbc->for_kupdate || wbc->for_background)
-		ret |= FLUSH_LOWPRI;
 	return ret;
 }
 
